node/src/chain_spec/mod.rs

```diff
@@ -193,7 +193,7 @@ pub fn testnet_genesis(
         .chain(initial_nominators.iter().map(|x| {
             use rand::{seq::SliceRandom, Rng};
             let limit = (MaxNominations::get() as usize).min(initial_authorities.len());
-            let count = rng.gen::<usize>() % limit;
+            let count = (rng.gen::<usize>() % limit).max(1); // at least one nomination
```

Comment:
How about:
```suggestion
            let count = (rng.gen::<usize>() % limit).saturating_add(1); // at least one nomination
```
The issue with the current approach is that `count` never reaches `initial_authorities.len()`, for example: with 2 initial authorities and 50 nominators, none of the nominates more than 1 authority. With 3 authorities and 50 nominators, none of them nominates more than 2 etc. This is because `rng.gen::<usize>() % limit` will never be equal to `limit`, but I don't think there is any reason it shouldn't be `== limit`

Reply:
Thats a great observation. I agree there isn't a reason why a nominator can't or shouldn't nominate all the initial validators. I think as long as we have something that works I'm happy. I doubt we are going to start a chain with a pre-defined set of nominators, let a lone randomly backing validators, but its good to have this functioning in anycase.